File: src/backend/Tmb.Shared/Models/Order.cs
namespace Tmb.Shared.Models;

public enum OrderStatus
{
    Pending,
    Processing,
    Completed
}

public class Order
{
    public Guid Id { get; set; }
    public string Customer { get; set; } = string.Empty;
    public string Product { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: src/frontend/app/core/models/Order.ts
export type OrderStatus = "Pending" | "Processing" | "Completed";

export type Order = {
  id: string;
  customer: string;
  product: string;
  amount: number;
  status: OrderStatus;
  createdAt: string;
};

export type CreateOrderDto = {
  customer: string;
  product: string;
  amount: number;
};